src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -309,10 +314,12 @@ public void setPeriod(String period) {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        if (containsSummaryTag(ast)) {
-            validateSummaryTag(ast);
+        boolean handled = false;
```

Comment:
logic should be still like:
```
if (isInlineTagPesentFrist(...)){
  validateInlineTag();
} esle {
if  (!starts.......)
}
```

this function is not heavy I think to  can afford it and recheck for summary and return sepaately
```
private static boolean isInlineTagPesentFrist(DetailNode javadoc) {
        final DetailNode[] child = javadoc.getChildren();

        // Checking size of ast is not required, since ast contains
        // children of Inline Tag, as at least 2 children will be present which are
        // RCURLY and LCURLY.
        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
                && ( RETURN_TEXT.equals(child[1].getText())
                           || SUMMARY_TEXT.equals.......);
    }
```

bu code will be easier to read.

Reply:
I think I updated it to be what you asked for?